File: Week_09/kotlin/src/_14_LongestCommonPrefix.kt
fun longestCommonPrefix(strs: Array<String>): String {
    if (strs.isEmpty()) return ""
    val firstRow = strs[0];
    var i = -1
    out@ while (++i < firstRow.length) {
        val c = firstRow[i]
        for (j in 1 until strs.size) {
            val s = strs[j];
            if (s.length == i || s[i] != c) break@out
        }
    }
    return firstRow.substring(0, i)
}

File: Week_09/cpp/src/_14_LongestCommonPrefix.cpp
#include <vector>
#include <string>
using namespace std;
class Solution
{
public:
    string longestCommonPrefix(vector<string> &strs)
    {
        // 按列比较
        if (strs.empty())
            return "";
        auto &first = strs[0];
        auto n = first.size(), m = strs.size();
        for (int i = 0; i < n; i++)
        {
            for (int j = 1; j < m; j++)
            {
                auto &curr = strs[j];
                if (i == curr.size() || first[i] != curr[i])
                {
                    return first.substr(0, i);
                }
            }
        }
        return first;
    }
};